Model/Api/RouteInsuranceManagement.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2022 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\Api;
+
+use Bolt\Boltpay\Helper\Bugsnag;
+use Exception;
+use Magento\Framework\Webapi\Exception as WebapiException;
+use Magento\Framework\Webapi\Rest\Response;
+use Magento\Framework\Module\Manager;
+use Magento\Framework\Serialize\SerializerInterface as Serialize;
+use Bolt\Boltpay\Helper\Cart as CartHelper;
+use Magento\Quote\Model\QuoteRepository;
+
+class RouteInsuranceManagement implements \Bolt\Boltpay\Api\RouteInsuranceManagementInterface
+{
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var Manager
+     */
+    private $moduleManager;
+
+    /**
+     * @var Serialize
+     */
+    private $serializer;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @var QuoteRepository
+     */
+    private $quoteRepository;
+
+    /**
+     * @param Response $response
+     * @param Bugsnag $bugsnag
+     * @param Manager $moduleManager
+     * @param Serialize $serializer
+     * @param CartHelper $cartHelper
+     * @param QuoteRepository $quoteRepository
+     */
+    public function __construct(
+        Response $response,
+        Bugsnag $bugsnag,
+        Manager $moduleManager,
+        Serialize $serializer,
+        CartHelper $cartHelper,
+        QuoteRepository $quoteRepository
+    ) {
+        $this->response = $response;
+        $this->bugsnag = $bugsnag;
+        $this->moduleManager = $moduleManager;
+        $this->serializer = $serializer;
+        $this->cartHelper = $cartHelper;
+        $this->quoteRepository = $quoteRepository;
+    }
+
+    /**
+     * @param $cartId
+     * @param $routeIsInsured
+     * @return void
+     * @throws WebapiException
+     */
+    public function execute($cartId, $routeIsInsured)
+    {
+        try {
+            if (!$this->isModuleEnabled()) {
+                $responseBody = [
+                    'message' => sprintf("%s is not installed on merchant's site", self::ROUTE_MODULE_NAME)
+                ];
+                $httpResponseCode = self::RESPONSE_FAIL_STATUS;
+            } else {
+                $responseBody = $this->collectTotalsAndSaveQuote($routeIsInsured, $cartId);
```

Comment:
Is it possible insted of using plugins update the session before this line?

Reply:
To update the session we'll need to know session_id from FE customer, this data is available in browser-server flow using cookie
In theory we can save session data in db for further use, but anyway it'll require to have any identifier of user to access correct session data. 
I think we need to discuss and investigate possibilities of supporting sessions through magento API more deeply, as it's big customization

Reply:
I mean even for API call we have session in place. And if we write into the session and initiate collect total route will read from this session.
Looks like simpler that approach with plugins